int/hwsc-user-svc/proto/hwsc-user-svc.proto

```diff
@@ -0,0 +1,203 @@
+    rpc DeleteUser (UserRequest) returns (UserResponse) {}
+
+
+    // chrome makes changes to a copy of User object (obtained when logged in), from user profile page
+    // gateway passes the modified User object
+    // svc will check that if the email changed, go through users documents to check for duplicated email
+    // svc will then do partial or complete update on users document?
+    // svc returns OK || email is already taken error || service is down error
+
+    // if OK, chrome sets offical User object with copy User object, displays success msg to user
+    // else, chrome will display error msg
+    rpc ModifyUserInfo (UserRequest) returns (UserResponse) {}
+
+
+    // chrome stores email/password in an object and makes request to login from login page
+    // gateway passes email/password object
+    // svc will look through user document for matching email
+    // svc will then match password (dehash, then match)
+    // svc will start session????
+    // svc returns OK with User object (with password nulled?) and some kind of session token?
+    //      || email doesn't match error
+    //      || if email match, password doesn't match error
+    //      || service is down error
+
+    // if OK, chrome will store user object and whatever token?, redirect to search page
+    // else, chrome will display error msg
+    rpc AuthenticateUserLogin (UserRequest) returns (UserResponse) {}
+
+
+
+
+    // THINGS GET KIND OF CONFUSING FROM HERE --------------------------------------
+
+
+    // chrome makes request to add friend by providing email? or by first and/or last name?
+    // gateway passes email (or first and last name)
+    // svc will look through user documents for matching email or name
+    // svc returns OK with found Users || email/name not found error || service is down error
+
+    // if OK, chrome will display Users first and last name and ask for confirmation?
+    //      || display list of matched names and corresponding emails and ask user to select correct one
+    //   Once user selects or confirms, chrome will store the matched users object under friends object
+    // else, chrome will display not found msg or error msg
+    rpc LookUpUser (UserRequest) returns (UserResponse) {}
+
+
+    // chrome makes request to share documents with found user
+    // gateway passes
+    //    1) request makers User object
+    //    2) a list of DUIDS to share with
+    //    3) found User object
+    // svc will...
+    rpc ShareDocuments (UserRequest) returns (UserResponse) {}
+
+
+    // chrome makes request to delete document/s
+    // gateway passes
+    //      1) document/s to delete
+    //      2) User object
+    // svc will go through users object documents map and delete key's (duid)
+    // svc will...
+    rpc DeleteDocuments (UserRequest) returns (UserResponse) {}
+
+}
+
+
+
+
+message DocumentData {
+    bool public = 1;
+    repeated string shared_with = 2;
```

Comment:
// Key = friend's uuid
// Value = This can be anything. This helps O(1) lookup time for deletion and updating
make this into  ``map<string, bool> shared_with = 2;``